nef/Preferences/ViewModels/PreferencesViewModel.swift

```diff
@@ -0,0 +1,150 @@
+    let fontItems:  [OptionItem]
+    let themeItems: [OptionItem]
+    let sizeItems:  [OptionItem]
+    
+    private let preferences: PreferencesDataSource
+    private let colors: [String: CarbonStyle.Color]
+    private let fonts: [CarbonStyle.Font]
+    private let themes: [CarbonStyle.Theme]
+    private let sizes: [CarbonStyle.Size]
+
+    init(preferences: PreferencesDataSource,
+         colors: [String: CarbonStyle.Color],
+         fonts: [CarbonStyle.Font],
+         themes: [CarbonStyle.Theme],
+         sizes: [CarbonStyle.Size]) {
+        
+        self.preferences = preferences
+        self.colors = colors
+        self.fonts = fonts
+        self.themes = themes
+        self.sizes = sizes
+        
+        let colorOptions = colors.keys.sorted().map { $0.itemColorName }.enumerated().map(OptionItem.init)
+        let customColorOption = OptionItem(id: colors.count, name: "-")
+        
+        self.state = preferences.state
+        self.colorItems = colorOptions + [customColorOption]
+        self.fontItems  = fonts.map { $0.itemName }.enumerated().map(OptionItem.init)
+        self.themeItems = themes.map { $0.itemName }.enumerated().map(OptionItem.init)
+        self.sizeItems  = sizes.map { $0.itemName }.enumerated().map(OptionItem.init)
+    }
+    
+    // MARK: internal attributes
+    private var currentFont: CarbonStyle.Font   { fonts.first(where:  { selectionFromFont($0)  == selectionFont  })! }
+    private var currentTheme: CarbonStyle.Theme { themes.first(where: { selectionFromTheme($0) == selectionTheme })! }
+    private var currentSize: CarbonStyle.Size   { sizes.first(where:  { selectionFromSize($0)  == selectionSize  })! }
+    private var currentColor: CarbonStyle.Color { colorFromHex ?? .nef }
+    
+    private var colorFromHex: CarbonStyle.Color? {
+        guard hex.count == 8 else { return nil }
+        return CarbonStyle.Color(hex: hex)
+    }
+    
+    private var hexFromColor: String? {
+        let selectedColor = colorItems[selectionColor].name.lowercased()
+        guard let selectedColorKey = colors.keys.first(where: { $0 == selectedColor }),
+              let color = colors[selectedColorKey] else { return colorFromHex?.hex }
+        
+        return color.hex
+    }
+    
+    // MARK: private methods
+    private func apply(state: PreferencesModel) {
+        self.showLines = state.showLines
+        self.showWatermark = state.showWatermark
+        self.selectionFont = selectionFromFont(state.font) ?? 0
+        self.selectionTheme = selectionFromTheme(state.theme) ?? 0
+        self.selectionSize = selectionFromSize(state.size) ?? 0
+        self.hex = state.color.hex
+    }
+    
+    private func persistState() {
+        preferences.persist(model: state)
+    }
+    
+    // MARK: helpers
+    private func selectionFromFont(_ font: CarbonStyle.Font) -> Int? {
+        return fontItems.enumerated().first(where: { $0.element.name == font.itemName })?.offset
```

Comment:
Why not implementing `Equatable` for `Font` instead of relying on checking names?

Reply:
yep, good point! :)